jpype/protocol.py

```diff
@@ -0,0 +1,92 @@
+import _jpype
+import datetime
+import sys
+import _jpype
+from . import _jclass
+from . import _jcustomizer
+
+# Copies of all private base types for reference
+_JClass = _jpype._JClass
+_JObject = _jpype._JObject
+_JException = _jpype._JException
+_JNumberLong = _jpype._JNumberLong
+_JNumberFloat = _jpype._JNumberFloat
+_JComparable = _jpype._JComparable
+_JChar = _jpype._JChar
+_JBoolean = _jpype._JBoolean
+_JArray = _jpype._JArray
+_JBuffer = _jpype._JBuffer
+
+
+if sys.version_info < (3, 8):
+    # Before 3.8 we have to use attribute type converters
+    from collections.abc import Sequence, Mapping
+
+    class _AttributeMeta(type):
+        def __instancecheck__(self, obj):
+            return hasattr(obj, self._attrib)
+    SupportsPath = _AttributeMeta("SupportsPath", (object,), {
+                                  "_attrib": "__fspath__"})
+    SupportsIndex = _AttributeMeta("SupportsIndex", (object,), {
+                                   "_attrib": "__index__"})
+    SupportsFloat = _AttributeMeta("SupportsFlaot", (object,), {
```

Comment:
Typo in first constructor argument (Fl**ao**t). Also I think SupportsFloat was there even before protocols, as of Python 3.5 (unlike SupportsIndex, that was introduced in 3.8).